robots/commenter/main.go

```diff
@@ -257,3 +308,45 @@ func run(c client, query, sort string, asc, random bool, commenter func(meta) (s
 	}
 	return nil
 }
+
+func getLastValidLabelsUpdate(ctx context.Context, client *github2.Client, issue github.Issue, goodLabels map[string]bool, badLabels map[string]bool) (time.Time, error) {
+	org, repo, number, err := parseHTMLURL(issue.HTMLURL)
+	if err != nil {
+		return time.Time{}, fmt.Errorf("error parsing HTML of PR %d: %v", issue.ID, err)
+	}
+	lastLabelDate := time.Time{}
+	for i := 1; true; i++ {
+		issueTimeline, _, err := client.Issues.ListIssueTimeline(ctx, org, repo, number, &github2.ListOptions{PerPage: 100, Page: i})
```

Comment:
should we maybe paginate through all events? after 100 it get's buggy right now (but I would put the pagination in another func)